### PR TITLE
Fallback on printing invalid wide string

### PR DESCRIPTION
**Pull request by cgzones**
In case the string to be printed contains invalid wide characters, fall
back to print as ASCII string.

Closes: https://github.com/htop-dev/htop/issues/1373

### CODE REVIEW

**Comment by BenBE** [comment]
Patch itself LGTM, but: Isn't there an multi-byte conversion function that uses an replacement character in case of invalid sequences? I.e. use U+FFFD for the erroneous sequence?

Instead of moving things around, we'd probably forward-declare the static functions at the top of the file.

**Comment by cgzones** [comment]
Used a different approach, retrying to interpret the remaining bytes still as multi-byte string.

File: RichString.c [hunk 1]
```diff
@@ -54,9 +54,45 @@ void RichString_rewind(RichString* this, int count) {
 
 #ifdef HAVE_LIBNCURSESW
 
+static size_t mbstowcs_nonfatal(wchar_t* dest, const char* src, size_t n) {
+   wchar_t *d = dest;
```
Comment (Explorer09) [inline RichString.c:58]: 1. Style
```suggestion
   wchar_t* d = dest;
```

2. I didn't get why you are duplicating two pointers `src` and `dest` with `s` and `d` respectively where you can just increment `src` and `dest` directly.

Comment (BenBE) [reply RichString.c:58]: Same goes for `remaining` vs `n`.

`s` and `d` are not very descriptive.

Also arguments are by-value, thus you can change the pointers locally without issues.

**Pull request merged**